### PR TITLE
Change Veranlagungszeitraum date

### PR DESCRIPTION
**Pull request by JulianRoesner**
# Short Description
This should adapt the tax period for which we are requesting the FSC. I had to adapt some of the tests. I tried to make them more reliable.

# Changes
- Change VERANLAGUNGSJAHR from 2020 to 2021
- Adapt tests

# Feedback
- Is that what you had in mind?
- What do you think about the adapted tests?

# How to review this Pull Request
[Link to Confluence](https://digitalservice4germany.atlassian.net/wiki/spaces/STL/pages/117637157/Development+Process#Approve-or-comment) (internal)

### CODE REVIEW

File: erica_app/erica/elster_xml/elster_xml_generator.py [hunk 1]
```diff
@@ -303,7 +303,7 @@ def _add_vast_list_xml_nutzdaten(xml_top, version='7'):
     list_xml.set('version', version)
 
 
-def _add_vast_beleg_ids_request_nutzdaten(xml_top, user_data, year='2020'):
+def _add_vast_beleg_ids_request_nutzdaten(xml_top, user_data, year='2021'):
```
Comment (frederike-ramin) [inline erica_app/erica/elster_xml/elster_xml_generator.py:306]: Oh wow, even adapted the beleg request, nice. 👍 I wouldn't have thought of that.

File: erica_app/tests/elster_xml/test_elster_xml_generator.py [hunk 2]
```diff
@@ -976,12 +980,13 @@ def test_add_vast_xml_nutzdaten_header(self):
...
-        _add_vast_request_xml_nutzdaten(xml_top, self.valid_user_data)
+        with patch('erica.elster_xml.elster_xml_generator.dt.date.today', MagicMock(return_value=datetime.date(_BEANTRAGUNGSJAHR, 1, 1))):
+            _add_vast_request_xml_nutzdaten(xml_top, self.valid_user_data)
         xml_string = _pretty(xml_top)
 
         self.assertEqual('false', xml_top.find('.//SpezRechtAntrag/Veranlagungszeitraum/Unbeschraenkt').text)
-        self.assertEqual('2020', xml_top.find('.//SpezRechtAntrag/Veranlagungszeitraum/Veranlagungsjahre/Jahr').text)
-        self.assertEqual('2021-12-31', xml_top.find('.//SpezRechtAntrag/GueltigBis').text)
+        self.assertEqual(str(VERANLAGUNGSJAHR), xml_top.find('.//SpezRechtAntrag/Veranlagungszeitraum/Veranlagungsjahre/Jahr').text)
+        self.assertEqual('2022-12-31', xml_top.find('.//SpezRechtAntrag/GueltigBis').text)
```
Comment (frederike-ramin) [inline erica_app/tests/elster_xml/test_elster_xml_generator.py:989]: Were you meaning to use the `_BEANTRAGUNGSJAHR` in this test instead of 2022?

**Review by frederike-ramin — APPROVED** [review]
Just one small comment, otherwise: Yes, that was what I had in mind. :)

File: erica_app/tests/elster_xml/test_elster_xml_generator.py (hunk 2 above)
Comment (JulianRoesner) [reply erica_app/tests/elster_xml/test_elster_xml_generator.py:989]: Good catch 👍

**Pull request merged**